src/Symfony/Bundle/SecurityBundle/Resources/views/Collector/security.html.twig

```diff
@@ -37,6 +37,13 @@
             <div class="sf-toolbar-info-piece">
                 <b>Actions</b>
                 <span><a href="{{ collector.logoutUrl }}">Logout</a></span>
+                {% if 'ROLE_PREVIOUS_ADMIN' in collector.roles %}
+                    - <span>
+                        <a href="{{ path(profile.getcollector('request').requestattributes.get('_route'), {'_switch_user': '_exit'}) }}">
```

Comment:
You're completely right (as usual), so I've been trying to retrieve these parameters but it seems I can't get any further than the `firewall` key. 
```php
$this->tokenStorage->getToken()->getProviderKey()
```
Seems that problem already was mentioned in http://stackoverflow.com/questions/19610316/how-to-access-firewall-configuration-in-controller

So, how can we "unlock" these configuration parameters for the `SecurityDataCollector`?